### PR TITLE
Update tier3.bicep Adding Defender configuration to Tier 3

### PR DESCRIPTION
**Pull request by LManning-Dev**
# Description

Added Defender Configuration used in mlz.bicep into the tier 3

## Issue reference

The issue this PR will close: #723 

## Checklist

Please make sure you've completed the relevant tasks for this PR out of the following list:

* [x] All acceptance criteria in the backlog item are met
* [x] The documentation is updated to cover any new or changed features
* [x] Manual tests have passed
* [x] Relevant issues are linked to this PR

### CODE REVIEW

**Comment by LManning-Dev** [comment]
Will do. Thank you

**Comment by LManning-Dev** [comment]
> Thank you for updating this. I propose that you also update the README located [here](https://github.com/Azure/missionlz/tree/main/src/bicep/add-ons/tier3) to include the additional optional parameter you are proposing be added to the Tier3 deployment.

@lisamurphy-msft 
I added the ability to pull the security option from the core (So it defaults to the same as the core but is overwritable) and added the variables to the readme. I think all 3 commits are showing up here.

Thank you for your time and help with this!

Note: Out of the 24 original variables, 22 are not in the readme. Should that be brought up in a different issue? 

**Parameters in the ReadMe**
param resourcePrefix string
param virtualNetworkAddressPrefix string = '10.0.125.0/26'

**Parameters not in the Readme**
param resourceSuffix string = 'mlz'
param location string = deployment().location
param workloadSubscriptionId string = subscription().subscriptionId
param mlzDeploymentVariables object = json(loadTextContent('../deploymentVariables.json'))
param hubSubscriptionId string = mlzDeploymentVariables.hub.Value.subscriptionId
param hubResourceGroupName string = mlzDeploymentVariables.hub.Value.resourceGroupName
param hubVirtualNetworkName string = mlzDeploymentVariables.hub.Value.virtualNetworkName
param hubVirtualNetworkResourceId string = mlzDeploymentVariables.hub.Value.virtualNetworkResourceId
param logAnalyticsWorkspaceResourceId string = mlzDeploymentVariables.logAnalyticsWorkspaceResourceId.Value
param logAnalyticsWorkspaceName string = mlzDeploymentVariables.logAnalyticsWorkspaceName.Value
param firewallPrivateIPAddress string = mlzDeploymentVariables.firewallPrivateIPAddress.Value
param deployDefender bool = mlzDeploymentVariables.deployDefender.Value
param emailSecurityContact string = mlzDeploymentVariables.emailSecurityContact.Value
param virtualNetworkDiagnosticsLogs array = []
param virtualNetworkDiagnosticsMetrics array = []
param networkSecurityGroupRules array = []
param networkSecurityGroupDiagnosticsLogs array
param networkSecurityGroupDiagnosticsMetrics array = []
param subnetAddressPrefix string = '10.0.125.0/27'
param subnetServiceEndpoints array = []
param logStorageSkuName string = 'Standard_GRS'
param tags object = {}

File: src/bicep/mlz.bicep [hunk 1]
```diff
@@ -1048,6 +1048,10 @@ output hub object = {
   networkSecurityGroupResourceId: hubNetwork.outputs.networkSecurityGroupResourceId
 }
 
+output deployDefender bool = deployDefender
```
Comment (lisamurphy-msft) [inline src/bicep/mlz.bicep:1051]: We are adding additional outputs to the MLZ baseline; this is to incorporate adding in outputs for the deploymentVariables.json for Tier3. 

While this isn't a big change, we should probably ensure that the outputs listed in the README.md of the MissionLZ bicep code include the outputs linked [in the outputs documentation](https://github.com/Azure/missionlz/tree/main/src/bicep#outputs) 

Please add to the list: 

deployDefender.value
emailSecurityContact.value

**Review by lisamurphy-msft — CHANGES_REQUESTED** [review]
Please address the comment included; this will slightly change the outputs listed in the documentation for a base MLZ deployment. While this is not a breaking change, it should be fixed to ensure accuracy. 

We are adding additional outputs to the MLZ baseline; this is to incorporate adding in outputs for the deploymentVariables.json for Tier3.

While this isn't a big change, we should probably ensure that the outputs listed in the README.md of the MissionLZ bicep code include the outputs linked [in the outputs documentation](https://github.com/Azure/missionlz/tree/main/src/bicep#outputs)

Please add to the list:

deployDefender.value
emailSecurityContact.value

**Review by lisamurphy-msft — APPROVED** [review]

**Comment by lisamurphy-msft** [comment]
Failing the Linting check; will investigate. 
Other then that I don't see any other issues preventing merging.

File: src/bicep/add-ons/tier3/README.md [hunk 2]
```diff
@@ -27,6 +27,8 @@ resourcePrefix | mlz | A prefix, 3 to 10 characters in length, to append to reso
 Optional Parameters | Default | Description
 ------------------- | ------- | -----------
 virtualNetworkAddressPrefix | 10.0.125.0/26 | The address prefix for the network spoke vnet.
+deployDefender | Output from mlz.bicep (false) | When set to "true", enables Microsoft Defender for Cloud for the subscriptions used in the deployment. It defaults to "false".
+emailSecurityContact | Output from mlz.bicep (empty '') | Email address of the contact, in the form of john@doe.com
```
Comment (lisamurphy-msft) [inline src/bicep/add-ons/tier3/README.md:31]: Let's remove or undo the changes here.
With the inheritance pattern proposed in this PR, this will not be necessary to explicitly state here. The state of just using the same default configuration in the original MLZ deployment will be sufficient with the added changes of documenting the additional two outputs.

**Comment by lisamurphy-msft** [comment]
Thank you for updating as requested; this looks good! 
Merging

**Review by lisamurphy-msft — APPROVED** [review]
